frontend/src/components/Dashboard/CustomAlert.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Alert, AlertTitle, Typography, Link } from '@mui/material';
+
+interface CustomAlertProps {
+  headerMsg?: string;
```

Comment:
Looks close. If you add the small test for the null alert, a guard in VSDashboardGate when no org id is available, and a unit test for isEmptyAfterScans. These are low-effort and reduce the risk of UX regressions around the new branch.

Reply:
Please see below commits per your requests on testing
- Tests to check if data is empty, null data scenarios, or present: 3426684090ce5452b68e32077095781767c55cdc
- Tests to check from the api call including when no org id is available: bb9feb25f8845d0b90c9b8d29f5c2668e1b73bbf
- Unit tests for isEmptyAfterScans and related utility functions: 9341f80ef9da54c385a25629c6e5325700b474c4